rdmd.d

```diff
@@ -713,9 +725,178 @@ private string[string] getDependencies(string rootModule, string workDir,
         exit(depsExitCode);
     }
 
+    if (!hasDependencies)
+        std.file.remove(rootModule);
+
     return dryRun ? null : readDepsFile();
 }
 
+bool needsInspection(string file)
+{
+    import std.utf : byChar;
+
+    foreach (match; file.matchAll(ctRegex!(`import[^a-zA-Z]*([^;]*);`, "s")))
+    {
+        // selective imports can only be at the end of an ImportList
+        // hence we can just chop of the imports
+        auto selectiveImportStart = match[1].byChar.countUntil(':');
+        if (selectiveImportStart < 0)
+            selectiveImportStart = match[1].length;
+
+        // the ImportList of packages can be separated with commas
+        foreach (importText; match[1][0..selectiveImportStart].splitter(","))
+        {
+            string mod;
+
+            // for renamed imports the second part is the package name
+            if (auto renamedImport = importText.matchFirst(ctRegex!(`[^=]*=\s*(.*)`, "s")))
+                mod = renamedImport[1];
+            else
+                mod = importText.strip;
+
+            // standard library and runtime are included by default and have no dependencies
+            if (!exclusions.any!(x => mod.startsWith(x.chain("."))))
```

Comment:
Thanks for the advice about using the `exclusions` array @aG0aep6G !

Is "chain" cheap enough or should I better allocate an `exclusionsWithDot` array at the top/start?